docs/usage.md

```diff
@@ -0,0 +1,323 @@
+# opentracing-cpp
+
+This file describes how clients can instrument their applications with the `opentracing-cpp`
+API. It assumes you have finished the required reading outlined [here](../README.md).
+
+### Outline
+
+The `opentracing-cpp` interface uses the [Curiously Recurring Template Pattern](https://en.wikipedia.org/wiki/Curiously_recurring_template_pattern)
+(CRTP) to define a set of compile time, polymorphic interfaces to `OpenTracing` implementations.
+
+In order to trace their systems, clients must:
+
+ - Select their `opentracing-cpp` implementation
+ - Install the selected `Tracer` in their applications in main
```

Comment:
I assumed there was only one, global tracer. My original implementation had a `Tracer::instance()` method that returned a pointer, but seeing `Tracer::instance()` for every call seemed redundant, so I wrapped the static methods around those calls.

This static interface is important for library developers.

The library developers don't have access to main, nor do they know about custom tracers being installed. The single point of entry to the `Tracer` removes that problem.

We could have an interface to return multiple tracers, but then all callers at all call sites then need to loop over instances. @lookfwd  had an elegant idea of higher order tracers, which we could install more primitive tracers into the top level global tracer. This would be particularly useful for transition from one implementation to another, but during the transition period, using both.

With all that being said, there's nothing stopping multiple instances of TracerImpl being stored independent or instead of the `instance()` method; you lose the singleton benefits, but gain multiple instances if thats a requirement.